source/elastic_solver.py

```diff
@@ -204,52 +336,80 @@ def _setup_boundary_conditions(self):
+    def set_dimensionless_numbers(self, C, D=None):
+        """
+        Updates the parameters of the model by creating or modifying class
+        objects.
+        """
+        assert isinstance(C, float)
+        assert isfinite(C)
+        assert C > 0.0
+        if not hasattr(self, "_C"):
+            self._C = dlfn.Constant(C)
+        else:
+            self._C.assign(C)
 
+        if D is not None:
+            assert isinstance(D, float)
+            assert isfinite(D)
+            assert D > 0.0
+            if not hasattr(self, "_D"):
+                self._D = dlfn.Constant(D)
+            else:
+                self._D.assign(D)
+    
     def _setup_problem(self):
         """
-        Method setting up non-linear solver objects of the stationary problem.
+        Method setting up nonlinear solver objects of the stationary problem.
         """
         assert hasattr(self, "_mesh")
         assert hasattr(self, "_boundary_markers")
 
         self._setup_function_spaces()
         self._setup_boundary_conditions()
 
-        # creating test and trial functions
-        u = dlfn.TrialFunction(self._Vh)
-        v = dlfn.TestFunction(self._Vh)
+        # creating test function
+        self._v = dlfn.TestFunction(self._Vh)
+        self._u = dlfn.TrialFunction(self._Vh)
 
         # solution
         self._solution = dlfn.Function(self._Vh)
 
         # volume element
-        dV = dlfn.Measure("dx", domain=self._mesh)
-        dA = dlfn.Measure("ds", domain=self._mesh, subdomain_data=self._boundary_markers)
-
-        # dimensionless parameters
-        C = self._C
+        self._dV = dlfn.Measure("dx", domain=self._mesh)
+        self._dA = dlfn.Measure("ds", domain=self._mesh, subdomain_data=self._boundary_markers)
         
-        # auxiliary function
-        def sym_grad(u):
-            return dlfn.Constant(0.5) * (grad(u) + grad(u).T)
+        # setup the parameters for the elastic law
+        self._elastic_law.set_parameters(self._mesh, self._C, self._u, self._v, self._solution)
         
-        # weak forms
-        # virtual work of internal forces
-        strain = sym_grad(u)
-        dstrain = sym_grad(v)
-        dw_int = (C * dlfn.tr(strain) * dlfn.tr(dstrain)
-                  + inner(dlfn.Constant(2.0) * strain, dstrain)
-                  ) * dV
+        # virtual work
+        self._dw_int = self._elastic_law.dw_int()
 
         # virtual work of external forces
-        dw_ext = dlfn.dot(self._null_vector, v) * dV
+        self._dw_ext = dlfn.dot(self._null_vector, self._v) * self._dV
 
         # add body force term
         if hasattr(self, "_body_force"):
             assert hasattr(self, "_D"), "Dimensionless parameter related to" + \
                                         "the body forces is not specified."
-            dw_ext += self._D * dot(self._body_force, v) * dV
-
+            self._dw_ext += self._D * self._elastic_law.volume_scaling() * dot(self._body_force, self._v) * self._dV
```

Comment:
I don't think that `self._elastic_law.volume_scaling()` but I am not sure. Please check this.